### DISCUSSION TITLE
Implementing Vertical Masonry Layout with Full Width Items

### DISCUSSION THREAD

**Discussion by BoggerByte** [TanStack/virtual#692] [Q&A]
I'm facing challenges in implementing a vertical masonry layout, especially when some items need to span the full width. Seeking insights and solutions from the community.

Any suggestions for achieving this layout while ensuring responsiveness and compatibility?

**Comment by 0XYoussefX0** [comment]
For people in the future, here is an example on how to virtualize a masonry layout : https://tanstack.com/virtual/latest/docs/framework/react/examples/variable
